client/src/App.jsx

```diff
@@ -8,7 +8,7 @@ const App = () => {
     <Layout>
       <Routes>
         <Route element={<HomePage></HomePage>} path="/" />
-        <Route element={<ProductPage></ProductPage>} path="/products" />
+        <Route element={<ProductPage></ProductPage>} path="/products/" />
```

Comment:
I think it's not necessary to add `/` at the end of route. `/products` should be enough.

Reply:
It removes `/` from URL after it redirects us to that page.